servicetalk-concurrent/src/main/java/io/servicetalk/concurrent/GracefulCloseable.java

```diff
@@ -15,10 +15,13 @@
  */
 package io.servicetalk.concurrent;
 
+import java.io.Closeable;
+import java.io.IOException;
+
 /**
- * An extension of {@link AutoCloseable} to add graceful closure semantics.
+ * An extension of {@link Closeable} to add graceful closure semantics.
  */
-public interface GracefulAutoCloseable extends AutoCloseable {
+public interface GracefulCloseable extends Closeable {
```

Comment:
Some notes from our offline discussion.

Since we are breaking API, I was considering whether the module owner of this class makes sense.
`GracefulCloseable` seems to be used in the context of I/O (ie. Http / gRPC). What is its correlation with the concurrent package? It would probably be a better fit, in some transport-related package.
WDYT? /cc @bondolo @idelpivnitskiy